src/nupic/regions/TMRegion.cpp

```diff
@@ -79,20 +79,57 @@ TMRegion::TMRegion(const ValueMap &params, Region *region)
 
   args_.iter = 0;
   args_.sequencePos = 0;
-  args_.outputWidth = (args_.orColumnOutputs)?args_.numberOfCols 
+  args_.outputWidth = (args_.orColumnOutputs)?args_.numberOfCols
                                              : (args_.numberOfCols * args_.cellsPerColumn);
   args_.init = false;
   tm_ = nullptr;
 }
 
-TMRegion::TMRegion(BundleIO &bundle, Region *region) : RegionImpl(region) {
+TMRegion::TMRegion(BundleIO &bundle, Region *region)
+    : RegionImpl(region), computeCallback_(nullptr) {
```

Comment:
can compute callback be moved to generic RegionImpl

Reply:
> can compute callback be moved to generic RegionImpl?

Yes, I was thinking about doing that.

Reply:
Oh, this must be used in the compute( ) method. In the base class compute( ) = 0; which forces the subclass to implement. So that is why I cannot push it down.